test/e2e/source_binding_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 The Knative Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or im
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+// +build !serving
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"gotest.tools/assert"
+	"knative.dev/client/pkg/util"
+)
+
+func TestSourceBinding(t *testing.T) {
+	t.Parallel()
+	test := NewE2eTest(t)
+	test.Setup(t)
+	defer test.Teardown(t)
+
+	test.serviceCreate(t, "testsvc0")
+
+	t.Run("create source binding", func(t *testing.T) {
+		test.sourceBindingCreate(t, "my-binding0", "Deployment:apps/v1:myapp", "svc:testsvc0")
+	})
+
+	t.Run("delete source binding", func(t *testing.T) {
+		test.sourceBindingDelete(t, "my-binding0")
+	})
+
+	t.Run("update source binding", func(t *testing.T) {
+		test.sourceBindingCreate(t, "my-binding1", "Deployment:apps/v1:myapp", "svc:testsvc0")
+		test.serviceCreate(t, "testsvc1")
+		test.sourceBindingUpdate(t, "my-binding1", "Deployment:apps/v1:myapp", "svc:testsvc1")
+		out, err := test.kn.RunWithOpts([]string{"source", "binding", "describe", "my-binding1"}, runOpts{NoNamespace: false})
+		assert.NilError(t, err)
+		assert.Check(t, strings.Contains(out, "testsvc1"))
+	})
```

Comment:
delete the services and binding created after the tests

Reply:
I mean the test resources: `testsvc0`, `testsvc1` and `my-binding1`

Reply:
The `tearDown` function will delete the namespace after test. I think it is unnecessary to delete these.

Reply:
Yes, :+1: